cluster-autoscaler/simulator/cluster.go

```diff
@@ -226,6 +226,7 @@ func (r *RemovalSimulator) findPlaceFor(removedNode string, pods []*apiv1.Pod, n
 			klog.Errorf("Simulating removal of %s/%s return error; %v", pod.Namespace, pod.Name, err)
 		}
 	}
+	r.clusterSnapshot.RemoveNodeInfo(removedNode)
```

Comment:
nit: Could you add a comment explaining that this is necessary for pod topology spread to work correctly?